tests/requests_compat/test_api.py

```diff
@@ -0,0 +1,30 @@
+"""Test compatibility with the Requests high-level API."""
+import pytest
+
+import requests
+
+
+@pytest.mark.copied_from(
+    "https://github.com/psf/requests/blob/v2.22.0/tests/test_requests.py#L61-L70"
+)
+def test_entrypoints():
+    requests.session
+    requests.session().get
+    requests.session().head
+    requests.get
+    requests.head
+    requests.put
+    requests.patch
+    requests.post
+
+
+@pytest.mark.copied_from(
+    "https://github.com/psf/requests/blob/v2.22.0/tests/test_requests.py#L72",
+    changes=["added noqa comment to silence flake8"],
+)
+@pytest.mark.xfail(
+    reason="PoolManager has no obvious equivalent in HTTPX", raises=ImportError
+)
+def test_poolmanager_entrypoint():
```

Comment:
What are your thoughts on straight up removing tests that have zero chance of running with HTTPX (like this test)?

Reply:
Besides this question this looks good to me :+1:

Reply:
So this relates to the “rigor” I mentioned in the PR description.

I think including these makes sure we track *important* Requests APIs that we don’t have. (In a sense, adding them and xfailing them is a form of testing too, in contrast with not including or skipping them?)

I don’t think it will be worth adding every single test from Requests though, as a lot of them test regressions or API details that may not be relevant to us.